### PR TITLE
[server] Parse incoming JWT sessions WEB-247

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Preview
* Sign-in
* See metrics endpoint contains count `k port-forward deployment/server 9500:9500, curl http://localhost:9500/metrics`

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-server-0ec62100c5</li>
	<li><b>🔗 URL</b> - <a href="https://mp-server-0ec62100c5.preview.gitpod-dev.com/workspaces" target="_blank">mp-server-0ec62100c5.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/session-handler.ts [hunk 1]
```diff
@@ -39,7 +42,31 @@ export class SessionHandlerProvider {
 
         options.store = this.createStore();
 
-        this.sessionHandler = session(options);
+        this.sessionHandler = (req, res, next) => {
+            let hasJWTCookie = false;
+
+            const cookies = parseCookieHeader(req.headers.cookie || "");
```
Comment (easyCZ) [inline components/server/src/session-handler.ts:48]: For some reason, `req.cookies` always contains the empty object, and never any of the actual cookies in the request. This is despite us using the `express-cookie` middleware which should populate it.

Doesn't matter much, as getting the token out of the header is simple enough, but it is odd.

Comment (geropl) [reply components/server/src/session-handler.ts:48]: Jup, this never worked. I think we are missing to set the first parameter `secret` of `cookieParser()` [here](https://github.com/gitpod-io/gitpod/blob/bcca71039ca01f0d833b487d036b1b03cb4c3335/components/server/src/server.ts#L155), to the same value we use [here](https://github.com/gitpod-io/gitpod/blob/233ec9457a99d935a8983e4229409ed652a76b2d/components/server/src/session-handler.ts#L37). 

Maybe worth playing around with.

File: components/server/src/session-handler.ts [hunk 2]
```diff
@@ -39,7 +42,31 @@ export class SessionHandlerProvider {
...
+            const jwtToken = cookies[SessionHandlerProvider.getJWTCookieName(this.config)];
+            if (jwtToken) {
+                // we handle the verification async, because we don't yet need to use it in the application
+                this.authJWT
```
Comment (geropl) [inline components/server/src/session-handler.ts:52]: I get a tslint error here:

```
/workspace/gitpod/components/server/src/session-handler.ts:52:17
ERROR: 52:17  no-floating-promises  Promises must be handled appropriately
```

But honestly no clue why that is the case, handling looks bullet-proof... :shrug: 

I'd say let's slap a `/* tslint:disable-next-line */` on here.

Comment (easyCZ) [reply components/server/src/session-handler.ts:53]: Added

File: components/server/src/session-handler.ts (hunk 1 above)
Comment (easyCZ) [reply components/server/src/session-handler.ts:48]: I'll investigate in another PR.

**Review by geropl — APPROVED** [review]
LGTM, tested and works! :heavy_check_mark:

**Pull request merged**